modules/ztp-preparing-the-hub-cluster-for-ztp.adoc

```diff
@@ -0,0 +1,101 @@
+//
+// *scalability_and_performance/ztp-zero-touch-provisioning.adoc
+
+[id="ztp-preparing-the-hub-cluster-for-ztp_{context}"]
+= Preparing the hub cluster for ZTP
+
+Use these procedures to configure your hub cluster with a set of ArgoCD applications that generate the required installation and policy CRs for each site based on a zero touch provisioning (ZTP) GitOps flow.
+
+.Procedure
+
+. Install the Red Hat OpenShift GitOps Operator on your hub cluster.
+
+. Extract the administrator password for ArgoCD:
++
+[source,terminal]
+----
+$ oc get secret openshift-gitops-cluster -n openshift-gitops -o jsonpath='{.data.admin\.password}' | base64 -d
+----
+
+. Prepare the ArgoCD pipeline configuration:
+.. Clone the Git repository.
+
+.. Modify the source values of the two ArgoCD applications, `deployment/clusters-app.yaml` and `deployment/policies-app.yaml` with appropriate URL, targetRevision (branch), and path values. The path values must match those used in your Git repository.
++
+Modify  `deployment/clusters-app.yaml`:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: Namespace
+metadata:
+     name: clusters-sub
+---
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+     name: clusters
+     namespace: openshift-gitops
+spec:
+    destination:
+         server: https://kubernetes.default.svc
+         namespace: clusters-sub
+  project: default
+  source:
+      path: ztp/gitops-subscriptions/argocd/resource-hook-example/siteconfig <1>
+      repoURL: https://github.com/openshift-kni/cnf-features-deploy <2>
+      targetRevision: master <3>
+  syncPolicy:
+      automated:
+          prune: true
+          selfHeal: true
+       syncOptions:
+       - CreateNamespace=true
+----
+<1> `path` is the branch that contains the `siteconfig` custom resources for the clusters.
+<2> `repoURL` is the URL of the Git repository that contains the `siteconfig` custom resources that define site configuration for installing clusters.
+<3> `targetRevision` is the branch on the Git repository that contains the relevant site configuration data.
+
+.. Modify `deployment/policies-app.yaml`:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: Namespace
+metadata:
+          name: policies-sub
+---
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+      name: policies
+      namespace: openshift-gitops
+spec:
+  destination:
+        server: https://kubernetes.default.svc
+        namespace: policies-sub
+  project: default
+  source:
+       directory:
+           recurse: true
+        path: ztp/gitops-subscriptions/argocd/resource-hook-example/policygentemplates <1>
+        repoURL: https://github.com/openshift-kni/cnf-features-deploy <2>
+        targetRevision: master <3>
+  syncPolicy:
+      automated:
+          prune: true
+          selfHeal: true
+       syncOptions:
+       - CreateNamespace=true
+----
+<1> `path` is the branch that contains the `policygentemplates` custom resources for the clusters.
```

Comment:
s/`policygentemplates` custom resources/`policygentemplates` CRs

Reply:
Earlier, `policygentemplates` and `siteconfig` were capitalized differently. Should that be made consistent throughout?

Reply:
I think in this case it's referring to the value shown in the yaml file.